src/main/java/org/codehaus/mojo/license/api/DefaultThirdPartyHelper.java

```diff
@@ -138,7 +139,7 @@ public DefaultThirdPartyHelper( MavenProject project, String encoding, boolean v
     {
         if ( artifactCache == null )
         {
-            artifactCache = new TreeMap<>();
+            artifactCache = Collections.synchronizedSortedMap( new TreeMap<String, MavenProject>() );
```

Comment:
I'd add double-checked locking:

wrap 142 like this:
synchronized(this); if(artifactCache==null) {
 --> initalization here
}

Reply:
@lama0206 interesting idea. Turns out `artifactCache` is static (yeah...), so synchronizing on `this` may not be enough. Could do this:
```suggestion
            synchronized (DefaultThirdPartyHelper.class)
            {
                if ( artifactCache == null )
                {
                    artifactCache = Collections.synchronizedSortedMap(new TreeMap<String, MavenProject>());
                }
            }
```

WDYT?

(It's been a long while since I looked at this code, so not sure whether this field _should_ be static. Can try to re-familiarize myself with the code later; likely not this evening.)

Reply:
Yep,  @Stephan202,  that should be it. Thanks for double-checking :)

Concerning _static_: #398 discusses the artifactCache being static (and never released/cleaned) and thus eating memory for the entire mvn build. I think this issue (#354) is about correctness and #398 is about memory consumption. So I'd say that this here is fine.

Reply:
Alright, rebased and updated the commit (squashed to keep history clean, since it appears in this repo squash-and-rebase isn't used much/at all).

Locally the `ISSUE-324/pom.xml` integration test fails, but that also happens on `master`. I was gonna say "let's see what happens on Travis CI", but it seems this repository hasn't been migrated to travis-ci.com yet, while travis-ci.org has ceased operations. Luckily there's also an AppVeyor build.)